archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -139,37 +138,39 @@ public void afterInjection(I injectee) {
             return injector.getInstance(Key.get(type, Names.named(name)));
         }
     }
+
+    private final AppConfig appConfig;
+    
+    public ArchaiusModule() {
+        this(null);
+    }
+    
+    public ArchaiusModule(AppConfig appConfig) {
+        this.appConfig = appConfig;
+    }
+    
+    public static ArchaiusModule fromAppConfig(final AppConfig config) {
+        return new ArchaiusModule(config);
+    }
     
     @Override
-    protected void configure() {
+    final protected void configure() {
         ConfigurationInjectingListener listener = new ConfigurationInjectingListener();
         requestInjection(listener);
-        
         bindListener(Matchers.any(), listener);
+        
+        bind(ConfigMapper.class).to(DefaultConfigMapper.class).in(Scopes.SINGLETON);
     }
     
     @Provides
     @Singleton
-    protected ConfigMapper createConfigMapper() {
```

Comment:
I haven't needed to do it yet, but I'm curious, how often is it expected that the user will  need to customize the `ConfigMapper`?